.github/workflows/improvedci.yml

```diff
@@ -0,0 +1,269 @@
+        permissions:
+            id-token: write
+            contents: read
+        steps:
+            - uses: actions/checkout@v4
+            - uses: actions/setup-node@v4
+              with:
+                  node-version-file: fronts-client/.nvmrc
+                  cache: yarn
+                  cache-dependency-path: fronts-client/yarn.lock
+            - run: yarn install --frozen-lockfile
+              working-directory: fronts-client
+            - run: yarn test
+              working-directory: fronts-client
+
+    integrationtest_client_v2:
+        runs-on: ubuntu-latest
+        permissions:
+            id-token: write
+            contents: read
+        needs:
+        - build_client_v2
+        steps:
+            - uses: actions/checkout@v4
+            - uses: actions/setup-node@v4
+              with:
+                  node-version-file: fronts-client/.nvmrc
+                  cache: yarn
+                  cache-dependency-path: fronts-client/yarn.lock
+            -   uses: actions/download-artifact@v4  #integration test requires built frontend (of course!)
+                with:
+                    path: public/fronts-client-v2
+                    name: fronts-client-v2
+            - run: yarn install --frozen-lockfile
+              working-directory: fronts-client
+            - run: yarn test-integration-ci
+              working-directory: fronts-client
+
+    test_backend:
+        runs-on: ubuntu-latest
+        permissions:
+            id-token: write
+            contents: read
+        services:
+            # See https://docs.github.com/en/actions/using-containerized-services/creating-postgresql-service-containers
+            postgres:
+                image: postgres:10.7-alpine
+                env:
+                    POSTGRES_USER: faciatool
+                    POSTGRES_PASSWORD: faciatool
+                    POSTGRES_DB: faciatool
+                # Set health checks to wait until postgres has started
+                options: >-
+                    --health-cmd pg_isready
+                    --health-interval 10s
+                    --health-timeout 5s
+                    --health-retries 5
+                ports:
+                    - 4724:5432
+        steps:
+            - uses: actions/checkout@v4
+            - uses: actions/setup-java@v4
+              with:
+                  java-version: "11"
+                  distribution: "corretto"
+                  cache: "sbt"
+            - name: Run tests
+              run: sbt 'test; database-int:test'
+
+    build_backend:
+        runs-on: ubuntu-latest
+        permissions:
+            id-token: write
+            contents: read
+        needs:
+            - build_client_v1
+            - build_client_v2
+            - generate_build_number
+        services:
+            # See https://docs.github.com/en/actions/using-containerized-services/creating-postgresql-service-containers
+            postgres:
```

Comment:
Is this necessary for the build? (No tests exercising a local DB?)

Reply:
Ah, probably not 😆